data_describe/anomaly_detection/detection.py

```diff
@@ -0,0 +1,548 @@
+            estimator: The anomaly detection estimator/model.
+            data_split (str, optional): The index to split the input data into a training set and testing set.
+                Note: Data is not shuffled.
+            viz_data (DataFrame): The data used for the default visualization.
+            input_data (DataFrame): The input data.
+            xlabel (str): The x-axis label for the anomaly plot.
+            ylabel (str): The y-axis label for the anomaly plot.
+            target (str, optional): The target column.
+            date_col (str, optional): The date column.
+            n_periods (int, optional): The number of periods for timeseries window.
+            **kwargs: Keyword arguments.
+        """
+        super(AnomalyDetectionWidget, self).__init__(**kwargs)
+        self.method = method
+        self.estimator = estimator
+        self.data_split = data_split
+        self.viz_data = viz_data
+        self.input_data = None
+        self.xlabel = None
+        self.ylabel = None
+        self.target = None
+        self.date_col = None
+        self.n_periods = None
+        self.sigma = None
+
+    def __str__(self):
+        return "data-describe Anomaly Detection Widget"
+
+    def __repr__(self):
+        return f"Anomaly Widget using {self.method}"
+
+    def show(self, viz_backend=None, **kwargs):
+        """The default display for this output.
+
+        Displays the anomalies, projected as a lineplot or scatterplot, with detected anomalies as red markers
+
+        Args:
+            viz_backend: The visualization backend.
+            **kwargs: Keyword arguments.
+
+        Raises:
+            ValueError: Data to visualize is missing / not calculated.
+
+        Returns:
+            The anomaly plot.
+        """
+        backend = viz_backend or self.viz_backend
+
+        if self.viz_data is None:
+            raise ValueError("Could not find data to visualize.")
+
+        return _get_viz_backend(backend).viz_plot_anomaly(
+            predictions_df=self.viz_data,
+            n_periods=self.n_periods,
+            xlabel=self.date_col,
+            ylabel=self.target,
+            **kwargs,
+        )
+
+
+def anomaly_detection(
+    data,
+    target: Optional[str] = None,
+    date_col: Optional[str] = None,
+    method: str = "arima",
+    estimator=None,
+    data_split: Optional[int] = None,
+    n_periods: Optional[int] = None,
+    sigma: float = 2.0,
+    compute_backend: Optional[str] = None,
+    viz_backend: Optional[str] = None,
+    **kwargs,
+):
+    """Identify and mark anamolies.
+
+    This feature identifies anomalies in timeseries and tabular data using multiple approaches (supervised, unsupervised, and statistical)
+    and then projects the data onto a plot for visualization.
+
+    Args:
+        data (DataFrame): The dataframe.
+        target (str, optional): The target column. Defaults to None. If target is None, unsupervised methods are used.
```

Comment:
this API won't be able to handle multivariate time series with this design? some additional thought required here for future iterations.